presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestPrestoDriver.java

```diff
@@ -1425,6 +1425,56 @@ public void testQueryCancellation()
         }
     }
 
+    @Test(timeOut = 4000)
+    public void testQueryTimeout()
+            throws Exception
+    {
+        try (Connection connection = createConnection("blackhole", "blackhole");
+                Statement statement = connection.createStatement()) {
+            statement.executeUpdate("CREATE TABLE test_query_timeout (key BIGINT) " +
+                    "WITH (" +
+                    "   split_count = 1, " +
+                    "   pages_per_split = 1, " +
+                    "   rows_per_page = 1, " +
+                    "   page_processing_delay = '1m'" +
+                    ")");
+        }
+
+        CountDownLatch queryFinished = new CountDownLatch(1);
+        AtomicReference<String> queryId = new AtomicReference<>();
+        AtomicReference<Throwable> queryFailure = new AtomicReference<>();
+
+        executorService.submit(() -> {
+            try (Connection connection = createConnection("blackhole", "default");
+                    Statement statement = connection.createStatement()) {
+                statement.setQueryTimeout(1);
+                try (ResultSet resultSet = statement.executeQuery("SELECT * FROM test_query_timeout")) {
+                    queryId.set(resultSet.unwrap(PrestoResultSet.class).getQueryId());
+                    try {
+                        resultSet.next();
+                    }
+                    catch (SQLException t) {
+                        queryFailure.set(t);
+                    }
+                    finally {
+                        queryFinished.countDown();
+                    }
+                }
+            }
+            return null;
+        });
+
+        // make sure the query timed out
+        assertTrue(queryFinished.await(2, SECONDS));
+        assertTrue(queryFailure.get().getMessage().contains("Query exceeded maximum time limit of 1.00s"));
```

Comment:
Use `assertContains` from Airlift `Assertions` as this will provide a better error message on failure